### PR TITLE
Redirect and messaging

### PR DESCRIPTION
**Pull request by tszarzynski**

### CODE REVIEW

File: src/worker/config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const TOKEN_NAME = 'token';
+export const GATEWAY_URL = '/gateway.html';
+export const INDEX_URL = '/test.html';
+
+export const WHITELISTED_URLS = [
+    '/gateway.html',
```
Comment (marcelkornblum) [inline src/worker/config.js]: shouldn't you use `GATEWAY_URL` here?

File: src/worker/config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const TOKEN_NAME = 'token';
+export const GATEWAY_URL = '/gateway.html';
+export const INDEX_URL = '/test.html';
+
+export const WHITELISTED_URLS = [
```
Comment (marcelkornblum) [inline src/worker/config.js]: `EXCLUDED_URLS` ? `BYPASSED_URLS` ?

File: src/worker/handlers.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    client.postMessage({ type: 'TOKEN_SET' });
+                });
+                break;
+            default:
+
+        }
+
+    }
+    else {
+        logger.log('[Service worker] Un-catched message:');
```
Comment (marcelkornblum) [inline src/worker/handlers.js]: Uncaught / unhandled :)

File: src/worker/handlers.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    return token;
+}
+
+/**
+ * Is request whitelisted
+ *
+ * @param {any} request
+ * @returns {Boolean}
+ */
+function isWhitelisted(request) {
```
Comment (marcelkornblum) [inline src/worker/handlers.js]: same as variable name above

File: src/worker/handlers.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+ * Handles Fetch Event
+ * 
+ * @export
+ * @param {any} event 
+ */
+export function handleFetch(event) {
+    logger.log(`[Service worker] Fetch event: ${event.request.url}`);
+
+    if (isAuthorized() || isWhitelisted(event.request)) {
+        if (isGateway(event.request)) {
```
Comment (marcelkornblum) [inline src/worker/handlers.js]: I don't understand this clause - is it possible for somethng to be whitelisted _and_ gateway? why would we redirect is authorized and gateway?

sorry if I'm being dumb

**Review by marcelkornblum — CHANGES_REQUESTED** [review]

File: src/worker/handlers.js (hunk 5 above)
Comment (tszarzynski) [reply src/worker/handlers.js]: We want redirect authorised users straight to index page if they are accessing gateway page.

File: src/worker/config.js (hunk 2 above)
Comment (tszarzynski) [reply src/worker/config.js]: Updated

File: src/worker/config.js (hunk 1 above)
Comment (tszarzynski) [reply src/worker/config.js]: Updated

File: src/worker/handlers.js (hunk 4 above)
Comment (tszarzynski) [reply src/worker/handlers.js]: Updated

File: src/worker/handlers.js (hunk 3 above)
Comment (tszarzynski) [reply src/worker/handlers.js]: Updated

**Review by marcelkornblum — APPROVED** [review]

**Pull request merged**